src/components/Snippet.jsx

```diff
@@ -1,34 +1,48 @@
-import React, { Component } from 'react';
+import React from 'react';
+import { connect } from 'react-redux';
 import { Controlled as CodeMirror } from 'react-codemirror2';
 
 import 'codemirror/lib/codemirror.css';
 
 import Title from './common/Title';
 import Input from './common/Input';
+import Spinner from './common/Spinner';
+import * as actions from '../actions';
 
 import '../styles/Snippet.styl';
 
-class Snippet extends Component {
-  constructor() {
-    super();
+class Snippet extends React.Component {
+  constructor(props) {
+    super(props);
     this.state = { isShowEmbed: false };
     this.toggleEmbed = this.toggleEmbed.bind(this);
   }
 
+  componentDidMount() {
+    const { dispatch } = this.props;
+    const { id } = this.props.match.params;
+
+    dispatch(actions.fetchSnippet(Number(id)));
+  }
+
   toggleEmbed() {
-    this.setState({ isShowEmbed: !this.state.isShowEmbed });
+    this.setState(prevState => ({ isShowEmbed: !prevState.isShowEmbed }));
   }
 
   render() {
+    const { snippet } = this.props;
+
+    if (!snippet) return <Spinner />;
+
     return (
       [
-        <Title title="Snippet" key="snippet" />,
-        <div className="snippet">
+        <Title title="Snippet" key="Snippet title" />,
+        <div className="snippet" key="Snippet">
           <div className="snippet-header">
             <div className="snippet-data">
               <div>
-                <span className="snippet-data-title">#235435, Untitled</span>
-                <span className="snippet-data-lang">[ Java ]</span>
+                <span className="snippet-data-title">#{snippet.get('id')}, {snippet.get('title')}</span>
```

Comment:
Could you please follow existing flow and show:

```
#snippet_id, Untitled
```

if `title` is `null`

and just title if not null.

Reply:
Yeap, sure. But notice that there should be same fix for recent items